app/src/main/java/ch/epfl/favo/view/tabs/MapsPage.java

```diff
@@ -124,32 +137,50 @@ public View onCreateView(
     FloatingActionButton button = view.findViewById(R.id.offline_map_button);
     button.setOnClickListener(this::onOfflineMapClick);
 
+    RadioButton toggle = view.findViewById(R.id.list_switch);
+    toggle.setOnClickListener(this::onToggleClick);
+    activity = (MainActivity) Objects.requireNonNull(getActivity());
+    setupView();
+    SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
+    if (mapFragment != null)
+      mapFragment.getMapAsync(this);
+    mLocation = DependencyFactory.getCurrentGpsTracker(getContext()).getLocation();
+    updateNearbyList();
     return view;
   }
 
-  @Override
-  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
-    super.onViewCreated(view, savedInstanceState);
-    setupView();
-    SupportMapFragment mapFragment =
-        (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
-    mGpsTracker = new GpsTracker(getContext());
-    if (mapFragment != null) {
-      mapFragment.getMapAsync(this);
+  private void updateNearbyList(){
+    String setting = activity.getPreferences(Context.MODE_PRIVATE).getString("radius", "1km");
```

Comment:
This is updated with respect to the current location focused on the screen right?

Reply:
right, current location is user's last location, updated every several minutes and retrieved when the map tab is switched to.

Reply:
Just wondering, how's the location updated every several minutes automatically?

Reply:
I don't understand exactly how does it work, but from my observation, every time you request a current Location, it actually return the last known location immediately. The last know location can be updated by two ways: the main way is by a listener onLocationChanged() that detects the location change of device, it reacts very quickly even if your phone changes position by several meters, the other way is requesting the update explicitly and the last known location will be updated about ten minutes later, this is almost useless because last know location is frequently overwritten by first way when you are moving. Above method is based on calling LocationManager, works well now,  there are other alternatives too.